### PR TITLE
Fix stripe public key retrieval

### PR DESCRIPTION
**Pull request by peschina**
### Description

Fix Stripe public key retrieval so we use provider key if valid. Fix secret key retrieval as well - we don't want to always default to our own key.

## Type

- [ ] **Dependency upgrade**
- [x] **Bug fix**
- [ ] **New feature**
- [ ] **Dev change**

### Checklist:

- [ ] I have added tests that prove my fix is effective or that my feature works

### Screenshots:

N/A

### CODE REVIEW

File: apps/payments/app/integration/stripe.ts [hunk 1]
```diff
@@ -19,27 +19,32 @@ const getSecretKey = async (
...
+    const stripe = await getStripeInstance(STRIPE_SECRET_KEY);
 
-export async function createPaymentIntent(paymentRequest: PaymentRequest) {
-  const providerId = getStripeProviderId(paymentRequest);
-  const stripe = await getStripeInstance(providerId);
-  return await stripe.paymentIntents.create({
-    amount: paymentRequest.amount,
-    currency: "EUR",
-  });
+    const paymentIntent = await stripe.paymentIntents.create({
```
Comment (nnorbert) [inline apps/payments/app/integration/stripe.ts:42]: Creating the intent could be extracted into a method.

**Review by nnorbert — APPROVED** [review]
Just one small comment. In rest it looks good!

File: apps/payments/app/integration/stripe.ts (hunk 1 above)
Comment (peschina) [reply apps/payments/app/integration/stripe.ts]: [4982bd1](https://github.com/ogcio/life-events/pull/163/commits/4982bd1ac1c126d4af7402fb17dceb5e52f1c6b6)

File: apps/payments/app/integration/stripe.ts [hunk 2]
```diff
@@ -19,27 +19,32 @@ const getSecretKey = async (
...
+
+export async function createPaymentIntent(paymentRequest: PaymentRequest) {
   try {
+    const providerId = getStripeProviderId(paymentRequest);
     const { userId } = await PgSessions.get();
     const sk = await getSecretKey(providerId, userId);
-    return new s(sk);
+    const stripe = await getStripeInstance(sk);
+
+    const paymentIntent = callCreateIntentApi(stripe, paymentRequest.amount);
```
Comment (nnorbert) [inline apps/payments/app/integration/stripe.ts]: callCreateIntent should be awaited.

File: apps/payments/app/integration/stripe.ts [hunk 3]
```diff
@@ -19,27 +19,32 @@ const getSecretKey = async (
...
+    const stripe = await getStripeInstance(STRIPE_SECRET_KEY);
 
-export async function createPaymentIntent(paymentRequest: PaymentRequest) {
-  const providerId = getStripeProviderId(paymentRequest);
-  const stripe = await getStripeInstance(providerId);
-  return await stripe.paymentIntents.create({
-    amount: paymentRequest.amount,
-    currency: "EUR",
-  });
+    const paymentIntent = callCreateIntentApi(stripe, paymentRequest.amount);
```
Comment (nnorbert) [inline apps/payments/app/integration/stripe.ts]: callCreateIntent should be awaited.

File: apps/payments/app/integration/stripe.ts [hunk 4]
```diff
@@ -19,27 +19,32 @@ const getSecretKey = async (
   return provider.data.liveSecretKey;
 };
 
-const getStripeInstance = async (providerId: string) => {
+const getStripeInstance = async (sk: string) => new s(sk);
+
+const callCreateIntentApi = async (stripe: s, amount: number) =>
+  await stripe.paymentIntents.create({
```
Comment (nnorbert) [inline apps/payments/app/integration/stripe.ts]: Here await is redundant, just return the result.

**Review by nnorbert — APPROVED** [review]

**Pull request merged**